.github/workflows/manual-sync.yml

```diff
@@ -0,0 +1,95 @@
+name: Sync from osu! upstream
+
+# Workflow dispatch can take inputs like any other github.event
+# As part of #5090 - we should evaluate adding inputs and outputs to refine this.
+on:
+  workflow_dispatch:
+    inputs:
+      backup:
+        description: "Create a backup of your target branch? (true/false)"
+        required: true
+        default: "false"
+      force:
+        description: "Overwrite any changes in the target repository? (true/false)"
+        required: true
+        default: "false"
+
+jobs:
+  sync:
+    name: Execute Sync Task
+    runs-on: ubuntu-latest
+    steps:
+     # WARNING: DO NOT USE SHALLOW CLONES FOR THIS STEP! Doing so will result on git merge thinking we diverged from upstream!
+      - name: Checkout
+        shell: bash
+        run: |
+          REPOSITORY="https://${{ github.actor }}:${{ github.token }}@github.com/${{ github.repository }}.git"
+          # merge commit ref name (with refs/heads/ stripped out)
+          BRANCH="${GITHUB_REF/#refs\/heads\//}"
+
+          git version
+          # clone without blobs; don't checkout to avoid fetching them anyway
+          git clone --filter=blob:none --no-checkout ${REPOSITORY} .
+          git config --local gc.auto 0
+
+          # set up sparse checkout
+          git sparse-checkout init
+          git sparse-checkout set '**/*.md' '**/.remark*' '**/*.json' '**/*.yaml'
+
+          git checkout --progress --force -B $BRANCH refs/remotes/origin/$BRANCH
+
+      - name: Configure Git to use your name
+        shell: bash
+        run: |
+          set -eo pipefail
+
+          git config --global user.name ${{ github.actor }}
+          git config --global user.email ${{ github.actor }}@users.noreply.github.com
+
+      - name: Begin Pull Task
+        shell: bash
+        run: |
+         set -eo pipefail
+
+         # sanitize github input into boolean
+         # https://stackoverflow.com/questions/14700579/bash-converting-string-to-boolean-variable
+         function boolean () {
+            case $1 in
+              true) echo true ;;
+              false) echo false ;;
+              *) echo "Unknown operand \"$1\"" 1>&2; exit 1 ;;
+            esac
+         }
+
+         #Upstream repository URL, this is where everything ties down.
+         UPSTREAM_REPO="https://github.com/ppy/osu-wiki"
+         BACKUP_BRANCH_NAME="backup-$(git branch --show-current)"
+         BRANCH_NAME="$(git branch --show-current)"
+
+         git remote add upstream $UPSTREAM_REPO
+         git fetch upstream
+
+         # User has no confidence in our git skills, so let's backup when needed!
+         if [ $(boolean ${{ github.event.inputs.backup }}) != false ]; then
+          git branch $BACKUP_BRANCH_NAME
+          git push -u origin +$BACKUP_BRANCH_NAME
+         fi
+
+         if [ $(boolean ${{ github.event.inputs.force }}) != true ]; then
+           git merge upstream/master --commit -m "Merge upstream/master changes to $BRANCH_NAME.";
```

Comment:
Does the merge commit message need to be specified here? I think the default would do just fine.

Reply:
Should be removed now in the latest rev.